src/EFCore.PG/Query/Expressions/Internal/ArrayAnyAllExpression.cs

```diff
@@ -95,7 +96,9 @@ public class ArrayAnyAllExpression : Expression, IEquatable<ArrayAnyAllExpressio
         protected override Expression Accept(ExpressionVisitor visitor)
             => visitor is NpgsqlQuerySqlGenerator npgsqlGenerator
                 ? npgsqlGenerator.VisitArrayAnyAll(this)
-                : base.Accept(visitor);
+                : visitor is NpgsqlSqlTranslatingExpressionVisitor npgsqlVisitor
```

Comment:
This seems very odd and non-standard... What's the idea?

Reply:
Looking at this again...I'm not sure.....but it makes things work?

I'll give this a closer look after the next commit. I suspect it's needed due to the translation/generation ~~changes~~ breakages I've introduced.

Reply:
Ah okay, I remember now:

Per some discussion in #551, this was a way to avoid overriding `VisitExtension(...)` on `NpgsqlSqlTranslatingExpressionVisitor`. 

But looking at it all again, I think we need to be handling this stuff in `VisitExtension(...)` after all. (The [default `SqlTranslatingExpressionVisitor`](https://github.com/aspnet/EntityFrameworkCore/blob/91653bdbe03cd2cbafc6c81c5bad8e1a7d4ef02f/src/EFCore.Relational/Query/ExpressionVisitors/SqlTranslatingExpressionVisitor.cs#L1077-L1144) seems to follow such a pattern.)